### DISCUSSION TITLE
QGIS-GRASS integration

### DISCUSSION THREAD

**Discussion by oskeng** [OSGeo/grass#2229] [Q&A] [answered]
I have built grass from source and would like to be able open grass data in qgis for quick visual inspection of results + cartography. I am on Fedora. 

The qgis-grass package will install grass78 + old wx so this is not an option. 

Any idea how this can be done?

**Comment by neteler** [comment]
Your QGIS installation needs to be sufficiently fresh.

GRASS GIS 8 support has been added in https://github.com/qgis/QGIS/pull/47095
i.e., updated in master (QGIS 3.24) and backported to QGIS 3.22 LTR.

I guess you could patch your local QGIS installation with this pull request.

In Fedora,
- GRASS GIS 8 is part of Fedora 36+: https://src.fedoraproject.org/rpms/grass
- QGIS 3.24 as well: https://src.fedoraproject.org/rpms/qgis

HTH, Markus

**Reply by oskeng** [reply]
Many thanks, @neteler.

I have now built both qgis and grass (and thus also wx) from source and they work individually. The grass plugin is activated by default in qgis, but when adding the gisdbase to favourites to browse the data, it is not recognized. Surely, I have missed something. Any advice are welcome.

EDIT: I have raised this issue in the QGIS community and will post here when I get it sorted out.

**Reply by oskeng** [reply]
Update:

Trying on Manjaro instead.

I have installed GRASS 8.0.1 from AUR and QGIS 3.24 from stable. 

grassprovider (2.12.99) is active but there is no toolbar and it is not possible to browse, or in any way identify, GRASS data. 

No replies from the QGIS community yet. Would really appreciate any hints in any (although preferably the right) direction.

@neteler?

**Reply by neteler** [reply]
> No replies from the QGIS community yet.

Where did you post it? Please cross-ref...

**Reply by oskeng** [reply]
> > No replies from the QGIS community yet.
> 
> Where did you post it? Please cross-ref...

https://gis.stackexchange.com/questions/425023/unable-to-access-grass-data-from-qgis

**Comment by neteler** [comment]
> I have built grass from source and would like to be able open grass data in qgis for quick visual inspection of results + cartography. I am on Fedora.

An option is to enable GRASS GIS data directly in GDAL/OGR using the respective drivers:
- raster: https://github.com/OSGeo/gdal/tree/master/frmts/grass
- vector: https://github.com/OSGeo/gdal/tree/master/ogr/ogrsf_frmts/grass

Here the related Fedora RPM package: [gdal-grass-plugin](https://copr.fedorainfracloud.org/coprs/neteler/gdal-grass-plugin/) for GDAL.

BTW: we also these drivers also here: https://mundialis.github.io/geoserver-grass-raster-datastore/

**Reply by oskeng** [reply]
> > I have built grass from source and would like to be able open grass data in qgis for quick visual inspection of results + cartography. I am on Fedora.
> 
> An option is to enable GRASS GIS data directly in GDAL/OGR using the respective drivers:
> 
>     * raster: https://github.com/OSGeo/gdal/tree/master/frmts/grass
> 
>     * vector: https://github.com/OSGeo/gdal/tree/master/ogr/ogrsf_frmts/grass
> 
> 
> Here the related Fedora RPM package: [gdal-grass-plugin](https://copr.fedorainfracloud.org/coprs/neteler/gdal-grass-plugin/) for GDAL.
> 
> BTW: we also these drivers also here: https://mundialis.github.io/geoserver-grass-raster-datastore/

Thanks a lot for helping out. 

Back on Manjaro where I feel more at home. Noted that gdal-grass plugin is available in AUR: https://aur.archlinux.org/packages/gdal-grass

I have installed, in this order: gdal, wx, grass, gdal-grass, qgis.

Still can't access GRASS data from QGIS.

Do I perhaps need to set a GISDBASE environment variable or something? 

Or am I perhaps missing something trivial? Can you please let me know how GRASS data could be accessed from QGIS, step-by-step, as you would expect it to work?

A bit embarrasing that I can't get this to work as I am fairly experienced in GRASS, QGIS, and Linux. Noteworthy is also that I haven't used GRASS-QGIS integration for a couple of years - I couldn't get it to work when I switched from Ubuntu to Arch and have run them in parallel ever since.

**Comment by neteler** [comment]
This is what I used (of course putting the path into the "Data Source Manager" is fine, too:

`qgis /home/mneteler/grassdata/nc_spm_08_grass7/PERMANENT/cellhd/elevation`

Note that I had to change to "Singleband pseudocolor" to actually see the map as it was initially white: 

![image](https://user-images.githubusercontent.com/1295172/157060598-b3200684-632e-46f1-8389-4c5129aa31a5.png)

**Comment by veroandreo** [comment]
This is how I see it in the browser pane, in case it's useful:

![image](https://user-images.githubusercontent.com/20075188/157064744-2d06c7cb-9d6a-44c1-95d6-188bc1f7bd73.png)

**Comment by oskeng** [comment]
Thank you both!

Now this was interesting. 

Running $ qgis /home/oskeng/Dropbox/Jobb/GIS/grassdata/Sweref99/Norrbotten/cellhd/NMD_buffer_final (cf. neteler) indeed works.

GRASS raster layers can also be added in the GUI using Layer>>Add layer>>Add raster layer. However, trying to open vector data the same way results in "Invalid data source: [...] is not a valid or recognized data source". 

Also, GRASS data is not recognized in the browser (cf. veroandreo). The same category as used above is shown as empty.

![Screenshot_20220308_092125](https://user-images.githubusercontent.com/61577586/157198692-15edbcb1-b99a-4de6-9567-656aee8c1c25.png)

**Comment by kikislater** [comment, marked as answer]
You need to rebuild grass then qgis if you want in browser, `gdal-grass` package will not help
You could use my Manjaro repo if you want : https://giscan.com/sig/pkg/ 

Edit : Use gdal-ecw available in repo, it have others options like full kml/kmz support unavailable in others packages even in AUR

![image](https://user-images.githubusercontent.com/13238064/157234644-b1338366-7b50-4d32-a9a4-f03cfaf9aef0.png)

**Reply by oskeng** [reply]
@kikislater, you are an absolute life saver. Installed grass and qgis from your repo and everything works. Many thanks!

Not able to install gdal-ecw though as it is in conflict with gdal. Any hints?

**Comment by kikislater** [comment]
gdal-ecw provides gdal so that's why it's in conflict. You should accept the removal of gdal in favor of gdal-ecw
Then ... 
```
$ pacman -Q gdal
gdal-ecw 3.4.1-2
```

**Reply by oskeng** [reply]
I get:
```
error: failed to prepare transaction (could not satisfy dependencies)
:: removing gdal breaks dependency 'gdal=3.4.0' required by python-gdal
```
Tried also installing gdal-ecw first but then I am prompted to replace gdal-ecw with gdal when installing QGIS.

**Comment by kikislater** [comment]
As you could see in original PKGBUILD (line 99) => https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=gdal-ecw#n99
`python-gdal-ecw` provides `python-gdal`

So you need to install `gdal-ecw` and `python-gdal-ecw`
